onnxruntime/core/providers/dml/DmlExecutionProvider/src/TensorDesc.cpp

```diff
@@ -315,3 +315,52 @@
     }
     m_bufferTensorDesc.DimensionCount = newDimensionCount;
 }
+
+// Uses dimensionMapping to reorder m_sizes and m_strides to match specific Tensor layout
+void TensorDesc::PermuteDimensions(const std::vector<uint32_t> dimensionMapping, const TensorAxis alignment)
```

Comment:
>const [](http://example.com/codeflow?start=81&length=5)

No need to make them const, because a function cannot modify the caller's state since it was passed by value. Also, no other functions do. e.g.

`SetDimensionCount(uint32_t newDimensionCount, TensorAxis alignment)`